### PR TITLE
FGHSYNC-2: Isolated connection errors from other type of errors.

### PR DESCRIPTION
**Pull request by mhawila**
Added a page displays transmission error history.
Added a task to remove old transmission error logs which is configurable via syn maintenance tab.

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/sync/api/SyncServiceTest.java [hunk 1]
```diff
@@ -92,7 +108,7 @@ public void deleteSyncRecords_shouldOnlyDeleteCommittedSyncRecordsIfChildNode()
     @Test
     @Verifies(value = "should exclude only types setup for all sync servers", method = "shouldSynchronize(Object)")
     public void shouldSynchronize_shouldOnlySyncValidTypes() throws Exception {
-    	executeDataSet("org/openmrs/module/sync/include/SyncServerClasses.xml");
+//    	executeDataSet("org/openmrs/module/sync/include/SyncServerClasses.xml");
```
Comment (steliomo) [inline api/src/test/java/org/openmrs/module/sync/api/SyncServiceTest.java]: Can this line be removed as it is commented?

File: api/src/main/java/org/openmrs/module/sync/SyncUtil.java [hunk 2]
```diff
@@ -1387,4 +1388,8 @@ public static String formatTransactionStatus(Transaction tx) {
 		}
 		return "TX STATUS UNKNOWN";
 	}
+
```
Comment (steliomo) [inline api/src/main/java/org/openmrs/module/sync/SyncUtil.java:1391]: Can you create a test for this method? According to the method it calculates the days between two dates. with a test scenarios will be much clear.

File: api/src/main/java/org/openmrs/module/sync/SyncUtil.java [hunk 3]
```diff
@@ -1387,4 +1388,8 @@ public static String formatTransactionStatus(Transaction tx) {
 		}
 		return "TX STATUS UNKNOWN";
 	}
+
+	public static int daysBetween(Date old, Date later){
+		return (int)( (later.getTime() - old.getTime()) / (1000 * 60 * 60 * 24));
```
Comment (steliomo) [inline api/src/main/java/org/openmrs/module/sync/SyncUtil.java]: You could also convert (1000 * 60 * 60 * 24) to a method with a clear name.
I know that it is converting a day right?

File: api/src/main/java/org/openmrs/module/sync/SyncUtilTransmission.java [hunk 4]
```diff
@@ -603,4 +565,60 @@ public static SyncTransmissionResponse doFullSynchronize(ReceivingSize size, Int
 			return response;
 		}
 	}
+
+	private static SyncTransmissionResponse transmitAndProcessResponse(RemoteServer server, IItem toTransmit) {
```
Comment (steliomo) [inline api/src/main/java/org/openmrs/module/sync/SyncUtilTransmission.java:569]: Can you please split this method dividing responsibilities to be more clear and also be easy to test.

**Review by steliomo — COMMENTED** [review]
@mhawila can you check my comments

File: api/src/test/java/org/openmrs/module/sync/api/SyncServiceTest.java (hunk 1 above)
Comment (mhawila) [reply api/src/test/java/org/openmrs/module/sync/api/SyncServiceTest.java]: Sure, I just forgot to remove it.

File: api/src/main/java/org/openmrs/module/sync/SyncUtil.java (hunk 2 above)
Comment (mhawila) [reply api/src/main/java/org/openmrs/module/sync/SyncUtil.java:1391]: Will do.

File: api/src/main/java/org/openmrs/module/sync/SyncUtil.java (hunk 3 above)
Comment (mhawila) [reply api/src/main/java/org/openmrs/module/sync/SyncUtil.java]: I will put the constants in the method itself.

File: api/src/main/java/org/openmrs/module/sync/SyncUtilTransmission.java (hunk 4 above)
Comment (mhawila) [reply api/src/main/java/org/openmrs/module/sync/SyncUtilTransmission.java:569]: No, sorry I can't this is not my code. We don't want to change stuff that is working.

**Pull request merged**